src/odemis/gui/comp/fastem_project_manager_panel.py

```diff
@@ -0,0 +1,1072 @@
+            0.3, 0.7
+        )  # Random lightness value between 0.3 and 0.7 for good visibility
+        new_saturation = random.uniform(
+            0.5, 1.0
+        )  # Random saturation value between 0.5 and 1.0 for vibrant colors
+        new_color_hls = (
+            new_hue,
+            new_lightness,
+            new_saturation,
+        )  # Fixed lightness and saturation
+        if is_unique(new_color_hls):
+            unique_color_found = True
+            new_color_rgb = hls_to_rgb(*new_color_hls)
+            new_color_hex = rgb_to_hex(new_color_rgb)
+            return new_color_hex
+
+
+class ProjectManagerImportExport:
+    """
+    Handles the import and export of project data for the FastEM project manager.
+
+    :param project_manager: (FastEMProjectManagerPanel) The project manager instance to import/export data for.
+    """
+
+    def __init__(self, project_manager):
+        self.project_manager = project_manager
+
+    def get_filepath_from_filedialog(
+        self, message: str, style: int
+    ) -> Union[str, None]:
+        """
+        Get the JSON file's path from wx.FileDialog.
+
+        :param: message: (str) The message to be shown in the filedialog.
+        :param: style: (int) The style of the filedialog.
+
+        :return: Union[str, None]: The chosen JSON file's path, if cancel is pressed return None.
+        """
+        with wx.FileDialog(
+            parent=self.project_manager.panel,
+            message=message,
+            wildcard="JSON files (*.json)|*.json",
+            style=style,
+        ) as filedialog:
+            if filedialog.ShowModal() == wx.ID_CANCEL:
+                return  # the user changed their mind
+            return filedialog.GetPath()
+
+    def export_to_file(self, filepath):
+        """
+        Exports the current sample's project data to a JSON file.
+
+        :param filepath: (str) The path to the file where data will be exported.
+        """
+        if not filepath.endswith(".json"):
+            filepath += ".json"
+        data = self._collect_project_data()
+        with open(filepath, "w") as file:
+            json.dump(data, file, indent=4)
+        wx.MessageBox(
+            f"Projects successfully exported for sample type {self.project_manager.main_data.current_sample.value.type}!",
+            "Export",
+            wx.OK | wx.ICON_INFORMATION,
+        )
+
+    def import_from_file(self, filepath):
+        """
+        Imports a sample's project data from a JSON file.
+
+        :param filepath: (str) The path to the file from which data will be imported.
+        """
+        with open(filepath, "r") as file:
+            data = json.load(file)
+        self._apply_project_data(data)
+        wx.MessageBox(
+            f"Projects successfully imported for sample type {self.project_manager.main_data.current_sample.value.type}!",
+            "Import",
+            wx.OK | wx.ICON_INFORMATION,
+        )
```

Comment:
Similarly, here, it shouldn't be necessary to have a blocking message box when everything went fine. Especially, the user should now see the data loaded. You could add a popup notification.
On the other hand, do show a message box with the exception message in case of error.